src/main/java/greymerk/roguelike/dungeon/settings/SpawnCriteria.java

```diff
@@ -59,37 +59,22 @@ public void setWeight(int weight) {
 
   public void setBiomeTypes(List<BiomeDictionary.Type> biomeTypes) {
     this.biomeTypes = biomeTypes;
-    this.everywhere = this.biomes.isEmpty() && this.biomeTypes.isEmpty();
   }
 
+  public boolean isValid(ISpawnContext spawnContext) {
+    return isEverywhere() || isBiomeValid(spawnContext) && isBiomeTypeValid(spawnContext);
+  }
 
-  public boolean isValid(ISpawnContext context) {
-
-    if (this.everywhere) {
-      return true;
-    }
-
-    boolean biomeFound = false;
-
-    if (this.biomes != null) {
-      biomeFound = context.includesBiome(biomes);
-    }
-
-    if (this.biomeTypes != null) {
-      biomeFound = context.includesBiomeType(this.biomeTypes);
-    }
+  private boolean isBiomeValid(ISpawnContext spawnContext) {
+    return biomes.isEmpty() || spawnContext.includesBiome(biomes);
+  }
 
-    return biomeFound;
+  private boolean isBiomeTypeValid(ISpawnContext spawnContext) {
+    return biomeTypes.isEmpty() || spawnContext.includesBiomeType(biomeTypes);
   }
```

Comment:
The function above (isValid) was really the problem. You'll notice in the old implementation, the "biomesFound" flag is overwritten if biomeTypes are provided. This meant that a user configuring the dungeon could only provide biomes or biomeTypes but never both, as the biomeTypes would always overrule the biomes.